src/bindings/c/include/openvino/c/ov_dimension.h

```diff
@@ -24,15 +30,15 @@ typedef struct ov_dimensions ov_dimensions_t;
  */
 
 /**
- * @brief Create a static dimension object
+ * @brief Init a static dimension object
  * @ingroup dimension
  * @param dimension_value The dimension value for this object
  * @param ov_status_e a status code.
  */
-OPENVINO_C_API(ov_status_e) ov_dimension_create(ov_dimension_t** dimension, int64_t dimension_value);
+OPENVINO_C_API(ov_status_e) ov_dimension_init(ov_dimension_t* dimension, int64_t dimension_value);
```

Comment:
I think we can keep the old name with create.
@riverlijunjie @ilya-lavrenov what do you think?

Reply:
Create() method will not permit user to create these structure by user application, but it is C structure and user should create this structure variable and pass its pointer to C APIs.

Reply:
ov_dimension_t is a completely C structure, user can allocate it memory directly, why still provide a method to allocate buffer for it? A init() method could provide user an interface to fill its data.

Reply:
We can remove create()/init() method from the simple C structure, such as dimension/shape/rank, but we should provide user a clear meaning to fill data for these structure.

Reply:
done - remove it due to it is unnecessary method.